h/services/mention.py

```diff
@@ -0,0 +1,53 @@
+import logging
+import re
+from typing import Iterable, Sequence
+
+import sqlalchemy as sa
+from sqlalchemy.orm import Session
+
+from h.models import Annotation, Mention
+from h.services.user import UserService
+
+USERID_PAT = re.compile(r"data-userid=\"([^\"]+)\"")
```

Comment:
Perhaps a bit too loose, but could work for prototyping

Reply:
In future, I think someone suggested the idea of extracting the ids from the annotation rendered HTML, via DOM selector (`html.find('a[data-hyp-mention]).dataset.userid`)

Reply:
Oh, I just noticed you commented precisely this here https://github.com/hypothesis/h/pull/9279#discussion_r1930436883